### PR TITLE
Offset batch: remove intermediary structure 

### PR DESCRIPTION
**Pull request by ennru**
## Purpose

Convert the `Map[GroupTopicPartition, OffsetAndMetadata]` directly into a `Map[TopicPartition, OffsetAndMetadata]` to be sent to the consumer actor.

## Changes

* Move the location where the `offset + 1` is applied
* Change `toString` back to just show the offsets

## Background Context

Before sending offsets in `KafkaAsyncConsumerCommitterRef` the map was first transformed to a `Seq[PartitionOffsetMetadata]` (unpacking the `OffsetWithMetaData`) and than back to a `Map[TopicPartition, OffsetAndMetadata]` in the `commit` method.
By mapping directly less intermediary objects are created. The `OffsetAndMetadata` can be kept, but the `offset + 1` needed to be applied when building the batch in `CommittableOffsetBatchImpl` instead. The public `offsets` method compensates for that.

### CODE REVIEW

File: core/src/main/scala/akka/kafka/internal/MessageBuilder.scala [hunk 1]
```diff
@@ -180,7 +180,7 @@ private[kafka] final class CommittableOffsetBatchImpl(
     }
 
     val newOffsets =
-      offsetsAndMetadata.updated(key, new OffsetAndMetadata(committableOffset.partitionOffset.offset, metadata))
+      offsetsAndMetadata.updated(key, new OffsetAndMetadata(committableOffset.partitionOffset.offset + 1L, metadata))
```
Comment (2m) [inline core/src/main/scala/akka/kafka/internal/MessageBuilder.scala]: I do not see immediately, why the change in offset increment and decrement above.

Comment (ennru) [reply core/src/main/scala/akka/kafka/internal/MessageBuilder.scala]: The `OffsetAndMetadata` is now passed through as-is. Before it was constructed, decomposed and constructed again with the `+ 1`.

File: core/src/main/scala/akka/kafka/internal/CommittableSources.scala [hunk 2]
```diff
@@ -139,26 +133,20 @@ private[kafka] class KafkaAsyncConsumerCommitterRef(consumerActor: ActorRef, com
       )
     )
 
-  private def commit(offsets: immutable.Seq[PartitionOffsetMetadata]): Future[Done] = {
-    val offsetsMap: Map[TopicPartition, OffsetAndMetadata] = offsets.map { offset =>
-      new TopicPartition(offset.key.topic, offset.key.partition) ->
-      new OffsetAndMetadata(offset.offset + 1, offset.metadata)
```
Comment (ennru) [inline core/src/main/scala/akka/kafka/internal/CommittableSources.scala]: Here was the `+ 1` before.
